src/Chapter2.hs

```diff
@@ -842,7 +865,9 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n xs | n < 0     = []
+            | otherwise = take (length xs) $ drop n $ cycle xs
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: